### PR TITLE
feat: collect more logs for Windows

### PR DESCRIPTION
**Pull request by AbelHu**
1. fix: fix network cleanup code on windows for contianerd nodes (#[4154](https://github.com/Azure/aks-engine/pull/4154))
2. fix: Adding csi-proxy logs to prevent access denied errors (#[4029](https://github.com/Azure/aks-engine/pull/4029))
3. feat: collect hyperv logs (#[3737](https://github.com/Azure/aks-engine/pull/3737))
4. feat: Collect calico logs

### CODE REVIEW

File: vhdbuilder/packer/test/vhd-content-test.ps1 [hunk 1]
```diff
@@ -109,24 +109,24 @@ function Test-FilesToCacheOnVHD
             if(![System.IO.File]::Exists($dest))
             {
                 Write-Error "File $dest does not exist"
-                $emptyFiles = $emptyFiles + $dest
+                $invalidFiles = $invalidFiles + $dest
                 continue
             }
+            $remoteFileSize = (Invoke-WebRequest $URL -Method Head).Headers.'Content-Length'
```
Comment (mainred) [inline vhdbuilder/packer/test/vhd-content-test.ps1:115]: 👍

File: vhdbuilder/scripts/windows/collect-windows-logs.ps1 [hunk 2]
```diff
@@ -1,6 +1,6 @@
 $ProgressPreference = "SilentlyContinue"
 
-$lockedFiles = "kubelet.err.log", "kubelet.log", "kubeproxy.log", "kubeproxy.err.log", "containerd.err.log", "containerd.log", "azure-vnet-telemetry.log", "azure-vnet.log"
+$lockedFiles = "kubelet.err.log", "kubelet.log", "kubeproxy.log", "kubeproxy.err.log", "azure-vnet-telemetry.log", "azure-vnet.log", "csi-proxy.log", "csi-proxy.err.log"
```
Comment (mainred) [inline vhdbuilder/scripts/windows/collect-windows-logs.ps1:3]: This is what we have in aks-engine master branch, do we need to include also `"network-interfaces.json", "interfaces.json"`?
```suggestion
$lockedFiles = "kubelet.err.log", "kubelet.log", "kubeproxy.log", "kubeproxy.err.log", "azure-vnet-telemetry.log", "azure-vnet.log", "network-interfaces.json", "interfaces.json", "csi-proxy.log", "csi-proxy.err.log"
```

**Review by mainred — APPROVED** [review]
Do we need to have full sync from aks-engine, and add calico part?

File: vhdbuilder/scripts/windows/collect-windows-logs.ps1 (hunk 2 above)
Comment (AbelHu) [reply vhdbuilder/scripts/windows/collect-windows-logs.ps1:3]: Those are for azure stack so I do not merge this commit

**Comment by AbelHu** [comment]
> Do we need to have full sync from aks-engine, and add calico part?

I have a branch for aks-engine to add calico support but is not finished yet

**Pull request merged**